Demurrage component (#471)
ergoplatform/ergo: papers/yellow/economy.tex

    Otherwise, the size of the state is always increasing with time, thus reducing clients performance.
\end{itemize}

To achieve this, we propose the following storage fee consensus rules.
Register $R3$ of a box contains tuple $(creation\_height, tx\_id || out\_num)$, where $creation\_height$ provided by a user
is used to determine the block height at the moment of transaction generation.
Transaction can only be put in the block of height $h$ if for every created box its $creation\_height \le h$.

Once the subsidized period for the box ends (that is,
$current\_block\_height \ge box.creation\_height + SP$, see $SP$ value below), anyone~(presumably, a miner) can
create a new box $out$ with the exactly the same content (including the guarding
script) except the monetary value and $R3$ content. The monetary value is to
reduced by $K \cdot B \cdot (out.creation\_height - self.creation\_height)$ maximum, where $B$ is the spent box~($self$)
size and $K$ is the storage cost of one byte. Thus, the difference is to be paid to the miner.
If box value is less that the storage fee, all the box content including tokens could be spent by the miner.

> For efficient lookup, we require a spending proof for an expired box to contain nothing but an index of an

Author: The text is corrected based on concrete feedback received via private discussion.

\end{itemize}

To achieve this, we propose the following storage fee consensus rules.
Register $R3$ of a box contains tuple $(creation\_height, tx\_id || out\_num)$, where $creation\_height$ provided by a user
is used to determine the block height at the moment of transaction generation.
Transaction can only be put in the block of height $h$ if for every created box its $creation\_height \le h$.

Once the subsidized period for the box ends (that is,
$current\_block\_height \ge box.creation\_height + SP$, see $SP$ value below), anyone~(presumably, a miner) can
create a new box $out$ with the exactly the same content (including the guarding
script) except the monetary value and $R3$ content. The monetary value is to
reduced by $K \cdot B \cdot (out.creation\_height - self.creation\_height)$ maximum, where $B$ is the spent box~($self$)
size and $K$ is the storage cost of one byte. Thus, the difference is to be paid to the miner.
If box value is less that the storage fee, all the box content including tokens could be spent by the miner.

For efficient lookup~(to find a proper output regarding an expired input without an iteration over transaction outputs),

> we require a spending proof for an expired box to be just a context extension which contains only an index of an

Reviewer: ```suggestion
... which contains only a variable with id $127$ containing an index of an output..
```

> we require a spending proof for an expired box to be just a context extension which contains only an index of an

Reviewer: From this 2 sentences it is still not clear, what is variable. I suppose to replace to something like:
```
...context extension which contains only a variable with id $127$ containing an index of an output  which is trying to spend the box.
```

output which is trying to spend the box. The variable identifier for the index in the extension is $127$.

We propose the following concrete parameters:
\begin{itemize}

>     \item{} $K$ - cost of storage of 1 byte of data in a State for 1 block.

Reviewer: ```suggestion
    \item{} $K$ - cost of storing 1 byte per Constants.StoragePeriod blocks, in nanoErgos.
```

>     Should be determined by miner votes, $12 (nanoErg/byte)$ by default, max value is 24.

Reviewer: `$12 (nanoErg/byte)$` is definitely incorrect now.
